Framework/DataHandling/src/LoadEventNexus.cpp

```diff
@@ -1695,78 +1686,48 @@ void LoadEventNexus::loadSampleDataISIScompatibility(
+ * that spectrum numbers in `indexInfo` argument are sorted and that the
+ * Parallel::StorageMode of `indexInfo` is `Cloned`. */
+Indexing::IndexInfo
+LoadEventNexus::filterIndexInfo(const Indexing::IndexInfo &indexInfo) {
+  // Spectrum numbers in indexInfo must be sorted.
+  // StorageMode must be cloned
 
-  // check if range [SpectrumMin, SpectrumMax] was supplied
   if (m_specMin != EMPTY_INT() || m_specMax != EMPTY_INT()) {
-    if (m_specMax == EMPTY_INT()) {
-      m_specMax = max;
-    }
-    if (m_specMin == EMPTY_INT()) {
-      m_specMin = min;
-    }
-
-    if (m_specMax > max) {
-      throw std::invalid_argument("Inconsistent range property: SpectrumMax is "
-                                  "larger than maximum spectrum found in "
-                                  "file.");
-    }
-
-    // Sanity checks for min/max
-    if (m_specMin > m_specMax) {
-      throw std::invalid_argument("Inconsistent range property: SpectrumMin is "
-                                  "larger than SpectrumMax.");
-    }
-
-    // Populate spec_list
+    // check if range [SpectrumMin, SpectrumMax] was supplied
+    if (m_specMax == EMPTY_INT())
+      m_specMax = static_cast<specnum_t>(
+          indexInfo.spectrumNumber(indexInfo.size() - 1));
+    if (m_specMin == EMPTY_INT())
+      m_specMin = static_cast<specnum_t>(indexInfo.spectrumNumber(0));
     for (int32_t i = m_specMin; i <= m_specMax; i++)
       m_specList.push_back(i);
-  } else {
-    // Check if SpectrumList was supplied
-
-    if (!m_specList.empty()) {
-      // Check no negative/zero numbers have been passed
-      auto itr = std::find_if(m_specList.begin(), m_specList.end(),
-                              std::bind2nd(std::less<int32_t>(), 1));
-      if (itr != m_specList.end()) {
-        throw std::invalid_argument(
-            "Negative/Zero SpectraList property encountered.");
-      }
-
-      // Check range and set m_specMax to maximum value in m_specList
-      if ((m_specMax =
-               *std::max_element(m_specList.begin(), m_specList.end())) >
-          *std::max_element(m_specList.begin(), m_specList.end())) {
-        throw std::invalid_argument("Inconsistent range property: SpectrumMax "
-                                    "is larger than number of spectra.");
-      }
-
-      // Set m_specMin to minimum value in m_specList
-      m_specMin = *std::min_element(m_specList.begin(), m_specList.end());
-    }
   }
-
   if (!m_specList.empty()) {
-
+    // Check if SpectrumList was supplied (or filled via min/max above)
+    const auto indices =
+        indexInfo.makeIndexSet(std::vector<Indexing::SpectrumNumber>(
+            m_specList.begin(), m_specList.end()));
+    m_specMin =
+        static_cast<specnum_t>(indexInfo.spectrumNumber(*indices.begin()));
+    m_specMax =
+        static_cast<specnum_t>(indexInfo.spectrumNumber(*(indices.end() - 1)));
+    const auto filteredIndexInfo = Indexing::extract(indexInfo, indices);
```

Comment:
A comment here explaining what is being filtered out to accompany the code would be useful. The fact that you are creating a complete slice as a new `IndexInfo` might not be so clear from the code alone.